etc/icinga2/scripts/mail-host-notification.sh

```diff
@@ -92,13 +92,33 @@ shift $((OPTIND - 1))
 
 ## Keep formatting in sync with mail-service-notification.sh
 for P in LONGDATETIME HOSTNAME HOSTDISPLAYNAME HOSTOUTPUT HOSTSTATE USEREMAIL NOTIFICATIONTYPE ; do
-	eval "PAR=\$${P}"
+  eval "PAR=\$${P}"
 
-	if [ ! "$PAR" ] ; then
-		Error "Required parameter '$P' is missing."
-	fi
+  if [ ! "$PAR" ] ; then
+    Error "Required parameter '$P' is missing."
+  fi
 done
 
+## Add line-breaks to very long host-outputs to avoid
+## mail servers rejecting the message because of hitting
+## a max. message line limit (RFC821 max. 1000b per line)
+##
+## but move on, if the strings seems to take care of its
+## own formating (containing \n or \r)
+if [ ! -z "${HOSTOUTPUT}" ] \
+   && [ "${#HOSTOUTPUT}" -ge 900 ] \
+   && ! [[ "${HOSTOUTPUT}" =~ ($'\n'|$'\r') ]]; then
+  TMP_OUTPUT=''
```

Comment:
Well, this would give those two scripts a dependency on awk - what they do not have at the moment. So far they only seem to rely on `mail` - which they obviously need - `cat`, `hostname` and `printf`. Everything else in these scripts are BASH-internals.

The if conditions above just:
- verify that the string is not empty
- it's line length is equal or greater than 900
- and the string does **not** contain a *carriage-return* `\r` or *newline* `\n` character
Only then it will take care of the line-length.

Nothing special so far, even compatible with BASH < v4